### ISSUE TITLE
Issue with "DragEvent.dataTransfer": Clarify that `dataTransfer` can be null.

### ISSUE THREAD

**Issue by jespertheend** [mdn/content#11670]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer

#### What information was incorrect, unhelpful, or incomplete?
I'm trying to figure out when the value of `dataTransfer` will be null. But I can't find anything about it in the spec.
Though in the typescript `lib.dom.d.ts` the property is marked as possibly null, and on mdn `if (ev.dataTransfer !== null)` is shown in the example. I was hoping to find more info on when it would be null on this page, or at least mention it being null at the return value section.

#### Specific section or headline?
https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer#return_value

#### What did you expect to see?
Mentioning that it could return null and when.

#### Did you test this? If so, how?
Not really, because I have no idea when dataTransfer will be null, but the spec mentions it possibly being null at https://html.spec.whatwg.org/multipage/dnd.html#the-dragevent-interface

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/dragevent/datatransfer`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/dragevent/datatransfer/index.md
* Last commit: https://github.com/mdn/content/commit/2279e5ae6c229c707a014a22aa1ec4635a0f981f
* Document last modified: 2021-09-14T18:08:09.000Z

</details>

**Comment by rohitb4** [comment]
@sideshowbarker can I work on this?

**Comment by hamishwillee** [comment]
@rohitb4 That would be great. You're assigned ;-)

**Comment by rohitb4** [comment]
I did try to figure out when `dataTransfer` object would be null. `dataTransfer` was not empty in many examples I tried and [spec](https://html.spec.whatwg.org/multipage/dnd.html#the-dragevent-interface) also doesn't mention when the value would be `null`.
However, I see some bugs in StackOverflow, which mentions while debugging Firefox did set the object to null. But that information is not conclusive.
